contributed_definitions/nyaml/NXsample_component_set.yaml

```diff
@@ -0,0 +1,37 @@
+category: base
+doc: |
+  Set of sample components and their configuration. The idea here is to store all descriptors that 
+  cannot be stored on the level of the individual sample component.
```

Comment:
This structure is definitely needed. I would call this NXSample_component and what you call NXsample_component is simply another NXsample or NXsubstance.

In this way you define samples and substances and then you can place them inside new samples as components decorated with concentrantion, volume fraction, etc.